plugin/src/main/kotlin/io/github/ttypic/swiftklib/gradle/task/CompileSwiftTask.kt

```diff
@@ -126,7 +129,7 @@ open class CompileSwiftTask @Inject constructor(
         "-Xswiftc",
         "-target",
         "-Xswiftc",
-        "${compileTarget.arch()}-apple-ios${minIos}.0${compileTarget.simulatorSuffix()}",
+        "${compileTarget.arch()}-apple-${platform(compileTarget, minIos, minMacosx)}.0${compileTarget.simulatorSuffix()}",
```

Comment:
@davidtaylor-juul do you know where we could find official documentation on what this triple can contain?

Reply:
I don't think there is any definitive documentation around what exact values to use, because it evolves constantly.  The documentation in the compiler (`xcrun swiftc -h`) simply says for `-target`: "Generate code for the given target <triple>, such as x86_64-apple-macos10.9" 

The clang compiler has a similar system and describes what the fields are a little more:
https://clang.llvm.org/docs/CrossCompilation.html#target-triple

And the source code parser has many of the actual values and logic:
https://github.com/apple/swift-driver/blob/main/Sources/SwiftDriver/Utilities/Triple.swift

Note that some are legacy and are conflated, so I don't think it's too important to be "official" here, but instead to use values that work with the current Swift toolchain, and cover all the targets that this plugin supports.

Reply:
Also, is `platform` a good name for the function here? Or is there better (standard) terminology?

Reply:
Thanks for all the links / helpful information!

Changed function name to `operatingSystem`.